src/serac/physics/operators/odes.cpp

```diff
@@ -25,27 +25,91 @@ void SecondOrderODE::SetTimestepper(const serac::TimestepMethod timestepper)
 {
   switch (timestepper) {
     case serac::TimestepMethod::HHTAlpha:
-      ode_solver_ = std::make_unique<mfem::HHTAlphaSolver>();
+      second_order_ode_solver_ = std::make_unique<mfem::HHTAlphaSolver>();
       break;
     case serac::TimestepMethod::WBZAlpha:
-      ode_solver_ = std::make_unique<mfem::WBZAlphaSolver>();
+      second_order_ode_solver_ = std::make_unique<mfem::WBZAlphaSolver>();
       break;
     case serac::TimestepMethod::AverageAcceleration:
-      ode_solver_ = std::make_unique<mfem::AverageAccelerationSolver>();
+      second_order_ode_solver_ = std::make_unique<mfem::AverageAccelerationSolver>();
       break;
     case serac::TimestepMethod::LinearAcceleration:
-      ode_solver_ = std::make_unique<mfem::LinearAccelerationSolver>();
+      second_order_ode_solver_ = std::make_unique<mfem::LinearAccelerationSolver>();
       break;
     case serac::TimestepMethod::CentralDifference:
-      ode_solver_ = std::make_unique<mfem::CentralDifferenceSolver>();
+      second_order_ode_solver_ = std::make_unique<mfem::CentralDifferenceSolver>();
       break;
     case serac::TimestepMethod::FoxGoodwin:
-      ode_solver_ = std::make_unique<mfem::FoxGoodwinSolver>();
+      second_order_ode_solver_ = std::make_unique<mfem::FoxGoodwinSolver>();
+      break;
+    case serac::TimestepMethod::NewmarkBeta:
+      second_order_ode_solver_ = std::make_unique<mfem::NewmarkSolver>();
+      break;
+    case serac::TimestepMethod::BackwardEuler:
+      first_order_system_ode_solver_ = std::make_unique<mfem::BackwardEulerSolver>();
       break;
     default:
       SLIC_ERROR("Timestep method was not a supported second-order ODE method");
   }
-  ode_solver_->Init(*this);
+
+  if (second_order_ode_solver_) {
+    second_order_ode_solver_->Init(*this);
+  } else if (first_order_system_ode_solver_) {
+    // we need to adjust the width of this operator
+    width *= 2;
+    first_order_system_ode_solver_->Init(*this);
+  }
+}
+
+void SecondOrderODE::Step(mfem::Vector& x, mfem::Vector& dxdt, double& t, double& dt)
+{
+  if (second_order_ode_solver_) {
+    // if we used a 2nd order method
+    second_order_ode_solver_->Step(x, dxdt, t, dt);
+  } else if (first_order_system_ode_solver_) {
```

Comment:
A else clause (or an extra one with a slic error) would prevent this from ever failing to define an ODE solver.

Reply:
I'll add a similar one for `FirstOrderODE`, I believe we don't check if `SetTimeStepper` has been called before running `Step`.